server/src/main/java/org/elasticsearch/repositories/blobstore/BlobStoreRepository.java

```diff
@@ -1845,9 +1859,9 @@ protected InputStream openSlice(long slice) throws IOException {
         }));
     }
 
-    private static ActionListener<Void> fileQueueListener(BlockingQueue<BlobStoreIndexShardSnapshot.FileInfo> files, int workers,
+    private static ActionListener<Void> fileQueueListener(BlockingQueue<BlobStoreIndexShardSnapshot.FileInfo> files,
                                                           ActionListener<Collection<Void>> listener) {
-        return ActionListener.delegateResponse(new GroupedActionListener<>(listener, workers), (l, e) -> {
+        return ActionListener.delegateResponse(new GroupedActionListener<>(listener, files.size()), (l, e) -> {
```

Comment:
I don't think this will work. If all files current in progress fail (say because of an abort or so), then you will never resolve this listener?